nrich-search/README.md

```diff
@@ -175,41 +201,167 @@ public class CarSearchService {
+| carTypeMake          | CAR_TYPE    |
+| carTypeModel         | CAR_MODEL   |
+
+would be consisted of a right join between entity `Car` and `CarType` (because of association relationship) and a predicate that would be similar to this SQL predicate.
+
+```sql
+WHERE manufacturedDate > 22-07-2022 AND manufacturedDate < 30-12-2022 AND price >= 15
+AND price < 20 AND numberOfKilometers = 100 AND make = 'CAR_TYPE' AND model = 'CAR_MODEL'
+```
+
+`CarSearchResult` matches results by property name and when including properties from association then `@Projection` annotation should be defined with path to property for correct mapping.
+
+```java
+    @Projection(path = "carType.make")
+    private final String carTypeMake;
+
+    @Value("carType.model")
+    private final String carTypeModel;
+```
+
+### Using `StringSearchExecutor` (quick search)
+
+`StringSearchExecutor`  accepts a query string, list of properties for search and `SearchConfiguration`.
+It is best used for **quick** search functionality.
+
+A custom repository interface for searching entities of type Car (same entity that we used in `SearchExecutor` chapter) is given bellow,
+
+```java
+public interface CarRepository extends JpaRepository<Car, Long>, StringSearchExecutor<Car> {
+
+}
+```
+
+The `StringSearchExecutor` query class is different from the `SearchExecutor` class in sense that it contains a search term and a property list.
+
+```java
+@Setter
+@Getter
+public class StringSearchCarDemoRequest {
+
+    @NotEmpty
+    private String searchTerm;
+
+    @Size(min = 1)
+    private List<String> propertyToSearchList;
+
+}
+```
+
+We can reuse the same result class `SearchCarResult` that was demonstrated in the `SearchExecutor` usage demonstration.
+
+In our service we would invoke it like this:
+
+```java
+
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+@Service
+public class CarSearchService {
+
+    private final CarRepository carRepository;
+
+    @Override
+    public Page<CarSearchResult> simpleSearch(StringSearchCarDemoRequest request) {
+        SearchConfiguration<Car, Car, Map<String, Object>> searchConfiguration = SearchConfiguration.<Car, Car, Map<String, Object>>builder()
+            .joinList(List.of(SearchJoin.innerJoinFetch("carType")))
             .resolvePropertyMappingUsingPrefix(true)
             .resultClass(CarSearchResult.class)
+            .anyMatch(true)
             .build();
 
-        return carSearchRepository.findAll(query, Arrays.asList("registrationNumber", "price"), searchConfiguration);
+        return carSearchDemoRepository.findAll(request.getSearchTerm(), request.getPropertyToSearchList(), searchConfiguration, Pageable.unpaged());
     }
 }
+```
 
+When using `simpleSearch` method we are searching the `Car` entity by properties supplied in `propertyToSearchList`.
+When properties is not a String type conversion is attempted using `StringToTypeConverter` and if it succeeds then property is searched otherwise it is ignored.
```

Comment:
I think it should be `when properties are not ...` or `when the property is not ...`.

Reply:
I meant `property`, in that case `is not` more correct.